charts/internal/seed-controlplane/charts/csi-driver-controller/templates/_deployment.tpl

```diff
@@ -0,0 +1,197 @@
+{{- define "csi-driver-controller.deployment" -}}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: csi-driver-controller-{{ .role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: csi
+    role: controller-{{ .role }}
+spec:
+  replicas: {{ .Values.replicas }}
+  revisionHistoryLimit: 0
+  selector:
+    matchLabels:
+      app: csi
+      role: controller-{{ .role }}
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+{{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      creationTimestamp: null
+      labels:
+        app: csi
+        role: controller-{{ .role }}
+        networking.gardener.cloud/to-dns: allowed
+        networking.gardener.cloud/to-public-networks: allowed
+        networking.gardener.cloud/to-shoot-apiserver: allowed
+    spec:
+      containers:
+      - name: csi-driver
+        image: {{ index .Values.images (print "csi-driver-" .role) }}
+        imagePullPolicy: IfNotPresent
+        args :
+        - --endpoint=$(CSI_ENDPOINT)
+        {{- if eq .role "file" }}
+        - --nodeid=dummy
+        {{- end }}
+        - --v=5
+        env:
+        - name: CSI_ENDPOINT
+          value: unix://{{ .Values.socketPath }}/csi.sock
+        - name: AZURE_CREDENTIAL_FILE
+          value: /etc/kubernetes/cloudprovider/cloudprovider.conf
+{{- if .Values.resources.driver }}
+        resources:
+{{ toYaml .Values.resources.driver | indent 10 }}
+{{- end }}
+        ports:
+        - name: healthz
+          containerPort: 9808
+          protocol: TCP
+        livenessProbe:
+          failureThreshold: 5
+          httpGet:
+            path: /healthz
+            port: healthz
+            scheme: HTTP
+          initialDelaySeconds: 10
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 3
+        volumeMounts:
+        - name: socket-dir
+          mountPath: {{ .Values.socketPath }}
+        - name: cloud-provider-config
+          mountPath: /etc/kubernetes/cloudprovider
+
+      - name: csi-provisioner
+        image: {{ index .Values.images "csi-provisioner" }}
+        imagePullPolicy: IfNotPresent
+        args:
+        - --csi-address=$(ADDRESS)
+        - --kubeconfig=/var/lib/csi-provisioner/kubeconfig
+        - --feature-gates=Topology=true
+        - --volume-name-prefix=pv-{{ .Release.Namespace }}
+        - --enable-leader-election
+        - --leader-election-type=leases
+        - --leader-election-namespace=kube-system
```

Comment:
I had to specify `--connection-timeout=15s` and `--timeout=120s` for successful dynamic pv creation. I saw the values from the example deploy from the driver repo - 
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/f6027051a9cca37b86d7007beeb6cf8e3dc3012d/deploy/v0.6.0/csi-azuredisk-controller.yaml#L29..L31

Otherwise with the default timeouts the provision fails with:
```
$ k describe pvc foo

  Normal   WaitForFirstConsumer  27m                   persistentvolume-controller                                                                          waiting for first consumer to be created before binding
  Normal   Provisioning          2m44s (x13 over 27m)  disk.csi.azure.com_csi-driver-controller-disk-58859754c5-qmb2c_ec703af3-a877-4d93-a4c7-179c1684b730  External provisioner is provisioning volume for claim "default/disk-claim"
  Warning  ProvisioningFailed    2m34s (x13 over 27m)  disk.csi.azure.com_csi-driver-controller-disk-58859754c5-qmb2c_ec703af3-a877-4d93-a4c7-179c1684b730  failed to provision volume with StorageClass "default": rpc error: code = DeadlineExceeded desc = context deadline exceeded
  Normal   ExternalProvisioning  2m8s (x102 over 27m)  persistentvolume-controller                                                                          waiting
```

```suggestion
        - --leader-election-namespace=kube-system
        - --connection-timeout=15s
        - --timeout=120s
```